internal/scaffold/cmd.go

```diff
@@ -127,11 +130,25 @@ func main() {
 		os.Exit(1)
 	}
 
-	// Create a new Cmd to provide shared dependencies and start components
-	mgr, err := manager.New(cfg, manager.Options{
+	// Set default manager options
+	options := manager.Options{
 		Namespace:          namespace,
+		MapperProvider:     restmapper.NewDynamicRESTMapper,
 		MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
-	})
+	}
+
+	// Add support for MultiNamespace set in WATCH_NAMESPACE (e.g ns1,ns2)
+	// Note that this is not intended to be used for excluding namespaces, this is better done via a Predicate 
+	// Also note that you may face performance issues when using this with a high number of namespaces.
+	// More Info: https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder
+	multiNamespace := strings.Split(namespace, ",")
+	if len(multiNamespace) > 1 {
+		options.Namespace = ""
```

Comment:
Doesn't this technically mean the operator watches all namespaces?

Reply:
Note that we added the default option above, then we need to overwrite in order to allow pass te MultiNamespacedCacheBuilder.

> options := manager.Options{
> 		Namespace:          namespace,

Reply:
Setting `options.Namespace = ""` _should_ be fine for multiple namespace caching, since its value is not used in that case.

Reply:
Closing it as solved. Tks for the help.